lib/index.es6

```diff
@@ -0,0 +1,34 @@
+import path from "path";
+import fs from "fs";
+
+import {sync as mkdirp} from "mkdirp";
+
+import compile from "./compile";
+import {entries} from "./util";
+
+export default function Interlock (options) {
+  const cwd = process.cwd();
+  options = options || {};
+
+  if (!options.emit || !options.emit.length) {
+    throw new Error("Must define at least one bundle.");
+  }
+
+  // TODO: validate options, or only persist specific values
```

Comment:
Do you know how it compares to Joi?  I planned to use that, mostly because I was familiar.  I don't mind heavy, so long as it isn't ridiculous.  Running in Node means I don't have to worry about page weight, etc.